packages/aragon-wrapper/src/index.js

```diff
@@ -348,59 +287,99 @@ export default class Aragon {
+      ]).then((values) => ({
+        kernelAddress: values[0],
+        appId: values[1],
+        codeAddress: values[2],
+        isForwarder: values[3]
+      }))
+    })
+
+    // Get all app proxy addresses
+    const baseApps$ = this.permissions.pipe(
       map(Object.keys),
       // Add Kernel as the first "app"
-      map((addresses) => {
-        const appsWithoutKernel = addresses.filter((address) => !this.isKernelAddress(address))
+      map((proxyAddresses) => {
+        const appsWithoutKernel = proxyAddresses.filter((address) => !this.isKernelAddress(address))
         return [this.kernelProxy.address].concat(appsWithoutKernel)
       }),
       // Get proxy values
       switchMap(
-        (appAddresses) => Promise.all(
-          appAddresses.map((app) => this.getProxyValues(app))
+        (proxyAddresses) => Promise.all(
+          proxyAddresses.map(async (proxyAddress) => {
+            let proxyValues
+            try {
+              proxyValues = await proxyContractValueCache.request(proxyAddress)
+            } catch (_) {}
+
+            return {
+              proxyAddress,
+              ...proxyValues
+            }
+          })
         )
       ),
-      map(appMetadata => appMetadata.filter(
-        (app) => this.isApp(app) || this.isKernelAddress(app.proxyAddress)
-      )),
-      // Get artifact info
-      flatMap(
+      // Filter to remove any non-apps assigned in permissions
+      map(appProxies => appProxies.filter(
+        (appProxy) => this.isApp(appProxy) || this.isKernelAddress(appProxy.proxyAddress)
+      ))
+    )
+
+    // Get artifact info for apps
+    const appsWithInfo$ = baseApps$.pipe(
+      switchMap(
         (apps) => Promise.all(
           apps.map(async (app) => {
-            if (!app.appId || !app.codeAddress) {
-              return app
-            }
-
-            const cacheKey = `${app.appId}.${app.codeAddress}`
-            const cachedAppInfo = dotprop.get(appInfoCache, cacheKey)
-
-            const appInfo =
-              cachedAppInfo ||
-              getAragonOsInternalAppInfo(app.appId) ||
-              (await this.apm
-                .getLatestVersionForContract(app.appId, app.codeAddress)
-                // Just silence any errors
-                .catch(() => { }))
-
-            if (!cachedAppInfo && appInfo) {
-              dotprop.set(appInfoCache, cacheKey, appInfo)
+            let appInfo
+            if (app.appId && app.codeAddress) {
+              const cacheKey = `${app.appId}.${app.codeAddress}`
+              try {
+                appInfo = await applicationInfoCache.request(cacheKey)
+              } catch (_) { }
             }
 
             return Object.assign(app, appInfo)
           })
         )
-      ),
-      // Replaying the last emitted value is necessary for this.apps' combineLatest to not rerun
-      // this entire operator chain on identifier emits (doing so causes unnecessary apm fetches)
```

Comment:
I think I was lying to myself when I made this comment before; `combineLatest()` only reduces emitted values so it should never rerun the entire operator chain. Maybe previously one of these observables wasn't hot (causing a new subscription chain to happen in the `combineLatest()`, but now all of them should be).